### PR TITLE
Add subcommand to run benchmarks

### PR DESCRIPTION
**Pull request by dureuill**
# Pull Request

## Related issue
Not user-facing, no issue

## What does this PR do?
- Adds a new `cargo xtask bench` subcommand that can run one or multiple workload files and report the results to a server
- A workload file is a JSON file with a specific schema
- Refactor our use of the `vergen` crate:
  - update to the beta `vergen-git2` crate
  - VERGEN_GIT_SEMVER_LIGHTWEIGHT => VERGEN_GIT_DESCRIBE
  - factor logic in a single `build-info` crate that is used both by meilisearch and xtask (prevents vergen variables from overriding themselves)
  - checked that defining the variables by hand when no git repo is available (docker build case) still works.
- Add CI to run `cargo xtask bench`

### CODE REVIEW

File: xtask/Cargo.toml [hunk 1]
```diff
@@ -11,5 +11,37 @@ license.workspace = true
...
+    "process",
+    "signal",
+] }
+tracing = "0.1.40"
+tracing-subscriber = "0.3.18"
+tracing-trace = { version = "0.1.0", path = "../tracing-trace" }
+uuid = { version = "1.7.0", features = ["v7", "serde"] }
+
+[build-dependencies]
+anyhow = "1.0.79"
```
Comment (irevoire) [inline xtask/Cargo.toml:47]: It’s already imported in the main dependencies

```suggestion
```

**Review by irevoire — CHANGES_REQUESTED** [review]

File: build-info/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum DescribeResult {
+    Prototype { name: &'static str },
+    Release { version: &'static str, major: u64, minor: u64, patch: u64 },
+    Prerelease { version: &'static str, major: u64, minor: u64, patch: u64, rc: u64 },
+    NotATag { describe: &'static str },
+}
```
[the remark this answers is not in the document]
Comment (dureuill) [reply build-info/src/lib.rs:34]: yup. the major and minor are not used yet, but that could come up in the dashboard

File: meilisearch/src/routes/mod.rs [hunk 3]
```diff
@@ -359,12 +359,18 @@ async fn get_version(
...
 
     HttpResponse::Ok().json(VersionResponse {
-        commit_sha: commit_sha.to_string(),
-        commit_date: commit_date.to_string(),
+        commit_sha: build_info.commit_sha1.unwrap_or("unknown").to_string(),
+        commit_date: build_info
+            .commit_timestamp
+            .and_then(|commit_timestamp| {
+                commit_timestamp
+                    .format(&time::format_description::well_known::Iso8601::DEFAULT)
```
[the remark this answers is not in the document]
Comment (dureuill) [reply meilisearch/src/routes/mod.rs:370]: 😞

**Review by irevoire — APPROVED** [review]
Perfecto, sorry for the `time` formatting error on my side 😅 

bors merge

**Pull request merged**